src/Pages/Goals.cshtml.cs

```diff
@@ -29,17 +30,20 @@ public class GoalsModel : PageModel
         public decimal Amount { get; set; }
 
         private GoalDatabase GoalDatabase { get; }
-        public GoalsModel(GoalDatabase goalDatabase)
+        private EntryDatabase EntryDatabase { get; }
+        public GoalsModel(GoalDatabase goalDatabase, EntryDatabase entryDatabase)
         {
             GoalDatabase = goalDatabase;
+            EntryDatabase = entryDatabase;
         }
 
         public async Task OnGet()
         {
             UserId = int.Parse(User.FindFirst(ClaimTypes.Name).Value);
-            var dataManager = new DataManager(UserId);
             Goals = await GoalDatabase.ReadListAsync(UserId);
-            Savings = dataManager.Income.EntryList.GetSum() - dataManager.Expenses.EntryList.GetSum();
+            var incomes = await EntryDatabase.ReadListAsync(UserId, EntryType.Income);
+            var expenses = await EntryDatabase.ReadListAsync(UserId, EntryType.Expense);
+            Savings = incomes.GetSum() - expenses.GetSum();
```

Comment:
This is where the dataManager was removed in goals :D

Reply:
Now it makes sense :D